pkg/query/remote_engine_test.go

```diff
@@ -24,6 +24,40 @@ import (
 	"github.com/thanos-io/thanos/pkg/store/labelpb"
 )
 
+func TestRemoteEndpoints_Engines(t *testing.T) {
+	t.Parallel()
+
+	clients := []Client{
+		NewClient(nil, "testclient1", nil),
+		NewClient(nil, "testclient2", nil),
+		NewClient(nil, "testclient3", nil),
+	}
+	endpoints := NewRemoteEndpoints(log.NewNopLogger(), func() []Client {
+		return clients
+	}, Opts{})
+
+	engines1 := endpoints.Engines()
+	testutil.Equals(t, len(clients), len(engines1))
+
+	// Compute lazy fields.
+	for _, e := range engines1 {
+		e.MaxT()
+		e.MinT()
+		e.LabelSets()
+		e.PartitionLabelSets()
+	}
+
+	engines2 := endpoints.Engines()
+	testutil.Equals(t, len(clients), len(engines2))
+
+	// endpoints.Engines() should return exactly the same slice.
+	testutil.Equals(t, len(engines1), len(engines2))
+	testutil.Equals(t, cap(engines1), cap(engines2))
```

Comment:
We can verify that the underlying slice data is exactly the same:

```go
testutil.Equals(t, unsafe.Pointer(unsafe.SliceData(engines1)), unsafe.Pointer(unsafe.SliceData(engines2)))
```

But I think, that's an overkill.

This would also be affected if the internal representation changes, e.g.:

```go
return slices.Clone(r.engines)
```